R/run-nmtran.R

```diff
@@ -0,0 +1,375 @@
+
+  # Combine NONMEM submission args
+  #  - The main ones of interest are prdefault, tprdefault, and maxlim, which
+  #    impact the evaluation of `NM-TRAN`
+  #  - Priority: .bbi_args > model yaml > bbi.yaml
+  cmd_args <- parse_nmtran_args(
+    .mod, .bbi_args = .bbi_args, nmfe_options = bbi_yaml$nmfe_options
+  )
+
+  # Check nonmem version
+  nm_config <- bbi_yaml$nonmem
+  if(is.null(nm_config)){
+    cli::cli_abort(
+      c(
+        "Did not find required NONMEM information in configuration file: {.path {bbi_yaml_path}}",
+        "i" = "Please see {.fun bbr::bbi_init} for additional details"
+      )
+    )
+  }
+  if (!is.null(.nonmem_version)) {
+    # check for valid version
+    if (!(.nonmem_version %in% names(nm_config))) {
+      cli::cli_abort(
+        c(
+          "Must specify a valid {.var nm_version} for {.fun bbr::run_nmtran}.",
+          "i" = "{.path {bbi_yaml_path}} contains the following options:",
+          "{names(nm_config)}",
+          "*" = "e.g., {.code run_nmtran(mod, .bbi_args = list(nm_version = '{names(nm_config)[1]}'))}"
+        )
+      )
+    }
+    default_nm <- nm_config[[.nonmem_version]]
+  }else{
+    # Look for default nonmem installation
+    default_nm <- purrr::keep(nm_config, function(nm_ver){
+      !is.null(nm_ver$default) && isTRUE(nm_ver$default)
+    })
+
+    if(length(default_nm) > 1){
+      nm_vers <- paste(names(default_nm), collapse = ", ")
+      cli::cli_abort(
+        c(
+          "Found multiple default NONMEM versions ({nm_vers}) at {.path {bbi_yaml_path}}",
+          "i" = "Please ensure only one version is set to the default"
+        )
+      )
+    }else if(length(default_nm) == 1){
+      default_nm <- default_nm[[1]]
+    }else{
+      # If no default, use the last one (likely higher version)
+      default_nm <- nm_config[[length(nm_config)]]
+    }
+  }
+
+  # Set NM-TRAN executable path
+  nm_path <- default_nm$home
+  nmtran_exe <- file.path(nm_path, "tr", "NMTRAN.exe")
+
+  # Check if nmtran_presort exists
+  nmtran_presort_exe <- file.path(nm_path, "util", "nmtran_presort")
+  run_presort <- unname(fs::file_exists(nmtran_presort_exe))
+
+  return(
+    list(
+      nmtran_exe = nmtran_exe,
+      nonmem_version = basename(nm_path),
+      cmd_args = cmd_args,
+      nmtran_presort_exe = if(run_presort) nmtran_presort_exe else NULL
+    )
+  )
+}
+
+#' Execute `NM-TRAN` in a given directory
+#'
+#' Execute `NM-TRAN` in a given directory. Also runs `nmtran_presort` if an
+#' executable is found.
+#'
+#' @param nmtran_exe Path to `NM-TRAN` executable.
+#' @param mod_path Path of a model to evaluate. Should be relative to `dir`. Nested
+#'  directories not supported.
+#' @param cmd_args A character vector of command line arguments for the `NM-TRAN`
+#'  execution call
+#' @param nmtran_presort_exe Path to `nmtran_presort` executable. Only available
+#' for `NONMEM` versions `nm74gf`, `nm74gf_nmfe`, and `nm75`. If provided, will run
```

Comment:
As I [mentioned in the last round](https://github.com/metrumresearchgroup/bbr/pull/705#discussion_r1693609994), I think it's confusing to use names like nm74gf and nm75 and so on in the documentation because these directories are Metworx-specific. It'd be better to just say the NONMEM version (e.g., "NONMEM 7.4 or later").

This function is not exported (and the docs are unlisted on the site due to the `internal` keyword), so this it's not a big deal if this stays as is.